From 92222edb9fe1a8bc2260062dff85c5fbffd2cd53 Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Thu, 9 Jun 2016 04:39:08 +0200
Subject: [PATCH 65/77] ar71xx/cpe510: enable LNA for CPE210/220/510/520

The LNA improves the rx path. Within a simple test setup
it improved the signal from -60dbm to -40dbm.

Signed-off-by: Alexander Couzens <lynxis@fe80.eu>

Backport of LEDE 94e23bf7409d6cc4c9efb55ed32aba8e5a497966
---
 target/linux/ar71xx/files/arch/mips/ath79/mach-cpe510.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/target/linux/ar71xx/files/arch/mips/ath79/mach-cpe510.c b/target/linux/ar71xx/files/arch/mips/ath79/mach-cpe510.c
index 8bf5c0f..5cb052a 100644
--- a/target/linux/ar71xx/files/arch/mips/ath79/mach-cpe510.c
+++ b/target/linux/ar71xx/files/arch/mips/ath79/mach-cpe510.c
@@ -30,6 +30,9 @@
 #define CPE510_GPIO_LED_L3	15
 #define CPE510_GPIO_LED_L4	16
 
+#define CPE510_GPIO_EXTERNAL_LNA0	18
+#define CPE510_GPIO_EXTERNAL_LNA1	19
+
 #define CPE510_GPIO_BTN_RESET	4
 
 #define CPE510_KEYS_POLL_INTERVAL	20 /* msecs */
@@ -93,6 +96,9 @@ static void __init cpe510_setup(void)
 					ARRAY_SIZE(cpe510_gpio_keys),
 					cpe510_gpio_keys);
 
+	ath79_wmac_set_ext_lna_gpio(0, CPE510_GPIO_EXTERNAL_LNA0);
+	ath79_wmac_set_ext_lna_gpio(1, CPE510_GPIO_EXTERNAL_LNA1);
+
 	ath79_register_m25p80(NULL);
 
 	ath79_register_mdio(1, 0);
-- 
2.10.2

